### PR TITLE
Add mocha test tool and create sample test folder structure

### PR DESCRIPTION
**Pull request by Wuseok**
I added mocha test tool and create test folder.
I refer to test-*.js in gulpfile.babel.js file.
and in case of each test subject, I'll create test-<subject>.js file and increasing about test case.
Could you check this request?

### CODE REVIEW

**Comment by Wuseok** [comment]
I add to test source code for node, npm, mongodb was installed or not.

File: test/test-start.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		execSync('./third_party/node/bin/npm -v', function(err, stdout, stderr){
+			if(err) assert(false);
+			else assert(true);
+		});
+	});
+});
+
+
+/*
+describe('Array test', function(){
```
Comment (romandev) [inline test/test-start.js]: Please remove this comments.

File: test/test-start.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+var assert = require("assert"); // assert module is supported from nodejs.
```
Comment (romandev) [inline test/test-start.js]: As we discussed offline, please use import instead of require().

File: gulpfile.babel.js [hunk 3]
```diff
@@ -77,3 +78,14 @@ gulp.task('stop', finish => {
     }, 1000);
   });
 });
+
+gulp.task('platform_test', () => {
```
Comment (romandev) [inline gulpfile.babel.js]: Let's rename platform_test with bootstrap_test

File: test/test-start.js (hunk 2 above)
Comment (romandev) [inline test/test-start.js]: Please add a boilerplate at head of this file as follows:
// Copyright (c) 2017 The Absolute Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

File: test/test-start.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+var assert = require("assert"); // assert module is supported from nodejs.
+const execSync = require('child_process').execSync;
+
+describe('Module check', function(){
```
Comment (romandev) [inline test/test-start.js]: Please do lint-check for this file.
This file doesn't follow our coding rule overall.

File: gulpfile.babel.js [hunk 5]
```diff
@@ -77,3 +78,14 @@ gulp.task('stop', finish => {
...
+  gulp.src(['test/test-*.js'], {read: false})
```
Comment (romandev) [inline gulpfile.babel.js]: I think the test file should be located in bootstrap/test/{test_name}.js.

File: test/test-start.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+var assert = require("assert"); // assert module is supported from nodejs.
+const execSync = require('child_process').execSync;
+
+describe('Module check', function(){
+	it('Is mongo db installed?', function() {
+		execSync('./third_party/mongodb/bin/mongod -version', function(err, stdout, stderr){
```
Comment (romandev) [inline test/test-start.js]: Is this really sync function? If so, why do we have to pass a callback to receive a result?

File: test/test-start.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+var assert = require("assert"); // assert module is supported from nodejs.
+const execSync = require('child_process').execSync;
+
+describe('Module check', function(){
+	it('Is mongo db installed?', function() {
```
Comment (romandev) [inline test/test-start.js]: We should use arrow function style.

File: test/test-start.js (hunk 2 above)
Comment (Wuseok) [inline test/test-start.js]: FIXME(Wuseok): babel is not build for test module.
Current babel was built in server part so It'll fix later.

Comment (Wuseok) [inline test/test-start.js]: I added Copyright here.

File: bootstrap/test/test-update.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+// found in the LICENSE file.
+
+var assert = require('assert');
+const execSync = require('child_process').execSync;
+// FIXME(Wuseok): babel is not build for test module.
+// Current babel was built in server part so It'll fix later.
+// please check github issue number #242
+
+describe('Update Test', () => {
+	it('Is update test running?', () => {
```
Comment (romandev) [inline bootstrap/test/test-update.js]: ditto.

File: bootstrap/test/test-update.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+var assert = require('assert');
+const execSync = require('child_process').execSync;
+// FIXME(Wuseok): babel is not build for test module.
```
Comment (romandev) [inline bootstrap/test/test-update.js]: ditto.

File: bootstrap/test/test-start.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+var assert = require('assert');
+const execSync = require('child_process').execSync;
+// FIXME(Wuseok): babel is not build for test module.
```
Comment (romandev) [inline bootstrap/test/test-start.js]: Please move this comment above |var assert = require('assert');|

File: bootstrap/test/test-start.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+var assert = require('assert');
```
Comment (romandev) [inline bootstrap/test/test-start.js]: Should be |const assert|

File: bootstrap/test/test-update.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2017 The Absolute Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+var assert = require('assert');
```
Comment (romandev) [inline bootstrap/test/test-update.js]: ditto.

File: bootstrap/test/test-update.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+var assert = require('assert');
+const execSync = require('child_process').execSync;
+// FIXME(Wuseok): babel is not build for test module.
+// Current babel was built in server part so It'll fix later.
+// please check github issue number #242
+
+describe('Update Test', () => {
+	it('Is update test running?', () => {
+		assert(true);
```
Comment (romandev) [inline bootstrap/test/test-update.js]: This is not implemented yet.
Please leave a FIXME or TODO comment.

**Comment by romandev** [comment]
@Wuseok, also please update .travis.yml.

Please see this reference: https://github.com/romandev/absolute/commit/0dca277a8a393af4a6d853f309c89ae131edce99

**Comment by Wuseok** [comment]
@romandev I applied about your comment. could you check again??
Thank you

**Comment by romandev** [comment]
This CL is too delayed. So, once merge this CL and then I'll update .travis.yml in follow-up patch.

**Pull request merged**